### PR TITLE
created a flag to update the counter

### PR DESCRIPTION
**Pull request by WesleyMPG**
The problem was counter value has been updated to the new value only when play button is toggled off. I just added a flag to make sure the counter is updated once when play button is toggle on.

I gonna be honest with you. In my experiênce i learnd gui should be fully separated from program functionality. This kind of architecture (gui mixed with base functions) in future can lead to really annoying maintainance and improvement problems. I really like this software and wathever you decide to do i'll try to help aways i can. And if you in some moment decide to refactor it i'll be pleasured to help.

### CODE REVIEW

**Comment by WesleyMPG** [comment]
In your gif i saw you set counter from 3 to 4 but only 2 repetitions were done. In my tests the previous value is the on to be executed.


Now it is ok.

File: atbswp/control.py [hunk 1]
```diff
@@ -361,6 +361,7 @@ class PlayCtrl:
     def __init__(self):
         self.count = settings.CONFIG.getint('DEFAULT', 'Repeat Count')
         self.infinite = settings.CONFIG.getboolean('DEFAULT', 'Infinite Playback')
+        self.cout_was_updated = False
```
Comment (RMPR) [inline atbswp/control.py:364]: Maybe you meant `count_was_updated` ? :)

**Comment by RMPR** [comment]
First of all thanks for the PR, I really appreciate.
Just tested and it seems okay, one problem though. Use the imperative 
for your commit message, if just take what you already have it would be
something like:

> Create a flag to update the counter

See [this blog post](https://chris.beams.io/posts/git-commit/) about the
rationale behind my change request.

Also if you can add a little description to the commit it would be
great. It will help for future maintenance and improvement ;)

> I gonna be honest with you. In my experiênce i learnd gui should be
> fully separated from program functionality. This kind of architecture
> (gui mixed with base functions) in future can lead to really annoying
> maintainance and improvement problems. I really like this software and
> wathever you decide to do i&#39;ll try to help aways i can. And if you
> in some moment decide to refactor it i&#39;ll be pleasured to help.
> You can view, comment on, or merge this pull request online at:

To be honest I tried to keep everything separated as much as possible.
gui for the GUI, control for the functionality and so on. The only exception 
as you have seen is the replay feature because of the threading thing. I
essentially give the control back to the main loop *in the gui module*
so I currently have no choice but to handle a tiny bit of the logic in
gui. I didn't see it as a big deal. I am open to discuss any idea you
have about that, feel free to open an issue or if you prefer I can PM
you to discuss more quickly, since issues can be a bit slow-paced.

File: atbswp/control.py (hunk 1 above)
Comment (WesleyMPG) [reply atbswp/control.py:364]: Yeah haha

**Pull request merged**